excelerator/imports/convert.js

```diff
@@ -312,140 +324,31 @@ function addToCache(dataCache, toUri, row, i, jobData, valFunc) {
     });
 }
 
-function getStatements(fromUri, isReverse, linksetUri) {
-    var toVar = "?to";
-    var wrappedUri = `<${fromUri}>`;
-    var subjectText = isReverse ? toVar : wrappedUri;
-    var objectText = isReverse ? wrappedUri : toVar;
-    //I feel this query it prone to breaking the second the datasource introduces unexpected relationships
-    //Like unexpected inferred relationships, for example transitiveOvelaps will break it and have to be explicitely removed
-    var query = `PREFIX void: <http://rdfs.org/ns/void#>
-PREFIX loci: <http://linked.data.gov.au/def/loci#>
-PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
-PREFIX geo: <http://www.opengis.net/ont/geosparql#>
-PREFIX geox: <http://linked.data.gov.au/def/geox#>
-PREFIX data: <http://linked.data.gov.au/def/datatype/> 
-PREFIX dct: <http://purl.org/dc/terms/>
-prefix dbp: <http://dbpedia.org/property/>
-PREFIX nv: <http://qudt.org/schema/qudt#numericValue>
-PREFIX qu: <http://qudt.org/schema/qudt#unit>
-PREFIX qb4st: <http://www.w3.org/ns/qb4st/>
-PREFIX epsg: <http://www.opengis.net/def/crs/EPSG/0/>
-
-SELECT distinct ?s ?to ?toParent (min(?subjectArea) as ?subjectAreaUnique) (min(?objectArea) as ?objectAreaUnique)
-WHERE {
-    ?s dct:isPartOf <${linksetUri}> ;
-       rdf:subject ${subjectText} ;
-       rdf:predicate ?p;
-       rdf:object ${objectText} .
-    OPTIONAL {
-        ${subjectText} geox:hasAreaM2 [ data:value ?subjectArea; qb4st:crs epsg:3577 ].  
-    }
-    OPTIONAL {
-        ${objectText} geox:hasAreaM2 [ data:value ?objectArea; qb4st:crs epsg:3577 ].  
-    }
-    OPTIONAL { FILTER (?toParent != ${wrappedUri})
-        ?s1 dct:isPartOf <${linksetUri}> ;
-            rdf:subject ?toParent ;
-            rdf:predicate geo:sfContains ;
-            rdf:object ?to .
-    }
-    FILTER (?p = geo:sfContains || ?p = geo:sfWithin)
- } group by ?s ?to ?p ?toParent
-`;
-
+export function getProportionStatements(fromUri, outputType, isCrosswalk = true, isContains = false, isWithin = false) {
     try {
-        // console.log(query)
-        var result = getQueryResults(query);
-        var json = JSON.parse(result.content);
-        var bindings = json.results.bindings;
-        var matches = bindings.map(b => {
-            if (b.toParent) {
-                var matchObj = {
-                    uri: b.toParent.value,
-                }
+        var result = HTTP.get(Meteor.settings.integrationApi.endpoint + "/location/overlaps", {
+            params: {
+                uri: fromUri,
+                areas: true,
+                proportion: true,
+                contains: isContains,
+                within: isWithin,
+                crosswalk: isCrosswalk,
+                output_type: outputType.uri,
+                output_dataset: outputType.datasetUri,
```

Comment:
this is interesting. i think this might be in a future scenario, where we have a featureType like MeshBlock but the dataset can be from either ASGS 2011, or ASGS 2016. Specifying the datasetUri might be valuable in that case.

like you mention, it wont cause errors, but at some point in the future, the location/overlaps endpoint might need a `output_dataset` param.

Reply:
Now that you mention it I do remember @shaneseaton remarking on this. As you say we can implement the corresponding API functionality in the future.

Reply:
I put this in a 'reminder' that it will be needed one day. Extra query paramaters are normally ignored so I figured it wouldn't break anything. It is there because when we eventually have two copies of GNAF or ASGS11 & 16, then we will need to differentiate which dataset we want Addresses, or Meshblock from.